brightnessd.c

```diff
@@ -1087,25 +1090,34 @@ static void spawn_async(char* cmd) {
 ///////////////////////////////////////////////////////////////////////////////
 // parse_cmd
 ///////////////////////////////////////////////////////////////////////////////
-/** Helper function to parse (and mutate) a cmd string into an exec-compatible array.
+/** Helper function to parse a cmd string into an exec-compatible array.
     Can deal with noisy whitespace.
 
-    @param cmd                the cmd string to be parsed, e.g. "/usr/bin/systemctl suspend"; variable will be mutated!
-    @param argv               the parse result, e.g. {"/usr/bin/systemctl", "suspend", NULL}
+    @param cmd                cmd string to be parsed, e.g. "/usr/bin/systemctl suspend"; variable will be mutated!
+    @param len                length of variable result
+    @param result             parse result, e.g. {"/usr/bin/systemctl", "suspend", NULL}
+    @return                   non-zero in case of an error
 */
-static void parse_cmd(char* cmd, char **argv) {
-    while (*cmd != (long) NULL) {
-        while (*cmd == ' ' || *cmd == '\t' || *cmd == '\n') {
+static int parse_cmd(const char* cmd, size_t len, char **result) {
+    size_t count = 0;
+    char* dup_cmd = strdup(cmd);
```

Comment:
Hm, why is `cmd` mutated as stated in the comments? You `strdup(cmd)` here and you're working on this copy from here onward. And, shouldn't this copy be `free()`d somewhere?

You may also squash this into the "Support for external commands on X11 events" commit...